File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.java
class Solution {
    public double minimumAverage(int[] nums) {
        Arrays.sort(nums);
        double ans=Integer.MAX_VALUE;
        int s=0, e=nums.length-1;
        while(s<e){
            double avg=(double)(nums[s]+nums[e])/2;
            if(ans>avg)ans=avg;
            s++;
            e--;
        }
        return ans;
    }
}

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        nums.sort()
        s,e=0,len(nums)-1
        ans=1000
        while(s<e):
            val=(nums[s]+nums[e])/2
            if val<ans:
                ans=val
            s+=1
            e-=1
        return ans